CI: TestConformance/MeshHTTPRouteMatching

```
--- FAIL: TestConformance/MeshHTTPRouteMatching (0.04s)

    conformance.go:72: 2025-06-27T03:29:19.008238819Z: Applying tests/mesh/httproute-matching.yaml
    apply.go:275: 2025-06-27T03:29:19.014296109Z: Creating mesh-matching HTTPRoute
    conformance.go:77: 2025-06-27T03:29:19.018873675Z: Running MeshHTTPRouteMatching, relying on the following features: Mesh-standard, HTTPRoute-standard
    pod.go:212: 2025-06-27T03:29:19.030377044Z: no pods found in app app=echo,version=v1
    apply.go:283: 2025-06-27T03:29:19.030434842Z: Deleting mesh-matching HTTPRoute

  --- FAIL: TestConformance/MeshHTTPRouteQueryParamMatching (0.07s)

```

https://github.com/cilium/cilium/actions/runs/15917333407/job/44897502828

/cc @cilium/sig-servicemesh

--- a/operator/pkg/gateway-api/conformance_test.go
+++ b/operator/pkg/gateway-api/conformance_test.go
@@ -72,11 +72,6 @@ func TestConformance(t *testing.T) {
 			})
 		}
 	}
-	// TODO: Run MeshGRPCRouteWeight once it is deflaked upstream. See
-	//       GH-42456 for details.
-	skipTests = append(skipTests, "MeshGRPCRouteWeight")
-	skipTests = append(skipTests, "MeshHTTPRouteMatching")  // same here
-	skipTests = append(skipTests, "MeshHTTPRouteNamedRule") // same here
 	options.TimeoutConfig.DefaultPollInterval = 1 * time.Second
 	options.UnusableNetworkAddresses = unusableNetworkAddresses
 	options.UsableNetworkAddresses = usableNetworkAddresses

